pkg/server/apiHandler.go

```diff
@@ -276,3 +276,42 @@ func (h *apiHandler) logs(w http.ResponseWriter, r *http.Request) {
 	// send back a 201
 	w.WriteHeader(http.StatusCreated)
 }
+
+func (h *apiHandler) uuid(w http.ResponseWriter, r *http.Request) {
+	uuidRequest, err := getClientUuidRequest(r)
+	if err != nil {
+		log.Printf("error getting uuidRequest: %v", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+	cert, err := parseX509Cert(uuidRequest.GetDeviceCert())
```

Comment:
Why are we using this new `parseX509Cert()` func, and not the existing `getClientCert()` that all of the others use? Is this different somehow?

Reply:
In this case, you need to take the certificate from the UuidRequest object, not from the request itself.

Reply:
Is that documented somewhere? Is this fundamentally different than all the other requests?

Reply:
In the V2 API I'd expect the full client cert to be in the authcontainer. If we also have it in the UuidRequest it means that for V2 we'd have it in two places; either they need to be checked to be identical or we don't use the UuidRequest DeviceCert (latter would make the message a lot smaller.)

If we support the uuid API endpoint in the V1 API it means we do need the DeviceCert in the uuid request.

Reply:
So, we need to add check that the TLS cert is the same with DeviceCert inside UuidRequest?

Reply:
I am not clear on this. Is `/uuid` v1 or v2? If it is v1, then the device cert needs to be available via mTLS. If it is v2, we should just defer on adding `/uuid` until adam goes v2.

Reply:
Somewhat related question, but for the other APIs (config, metrics, info) where we have a UUID in the URL and one which we fetch based on the cert, does Adam/Eden check that they match?